runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -185,7 +185,8 @@ public Token appendCheckpoint(Token snapshotTimestamp, Long streamTail) {
     private Token forceNoOpEntry() {
         TokenResponse writeToken = rt.getSequencerView().next(streamId);
         LogData logData = new LogData(DataType.HOLE);
-        rt.getAddressSpaceView().write(writeToken, logData, CacheOption.WRITE_AROUND);
+        logData.useToken(writeToken);
+        rt.getAddressSpaceView().write(logData, CacheOption.WRITE_AROUND);
```

Comment:
Fail the write method if a token hasn't been set ? (check Address.NON_ADDRESS ?)